feat(native): add traces_sampler support (#12333)
getsentry/sentry-docs: platform-includes/performance/default-sampling-context-platform/native.mdx

> For the native SDK, it includes a [Transaction Context](https://github.com/getsentry/sentry-native/blob/5cb09867668a2b1d09a13a2a13253a83127a31d9/src/sentry_tracing.c), a custom sampling context (`sentry_value_t` which can be used to store 'char *' keys with `sentry_value_t` values (from `int32`, `bool`, `double` or `char *`)) and a `parent_sampled` flag (`int *`) indicating whether there was a sampling decision in the transaction's parent (if it is `NULL`, there is no decision. Pointing to `0` indicates it wasn't sampled, and to a `1` means it was).

Reviewer: Only a suggestion: we could go a bit lighter on the eye and lose a bit of detail here. A list and removing the details from the `sentry_value_t` might make this a bit easier to read: 

```suggestion
For the native SDK, it includes 
* a [Transaction Context](https://github.com/getsentry/sentry-native/blob/5cb09867668a2b1d09a13a2a13253a83127a31d9/src/sentry_tracing.c), 
* a custom sampling context (a `sentry_value_t` of arbitrary structure) and
* a `parent_sampled` flag (`int*`) indicating there wasn't a sampling decision in the transaction's parent if it is `NULL`.  If the pointer isn't `NULL`, a value of `0` indicates the parent wasn't sampled, and a `1` means it was.
```

But I am not 100% on that either.

Author: Makes sense, in the existing docs (e.g. for [Apple](https://docs.sentry.io/platforms/apple/configuration/sampling/#sampling-context-data)) this section was rather short, but since it needs some more explanation on Native I agree that splitting it up into bullet points & keeping it as simple as possible is way easier to read.
